src/algebra/group/basic.lean

```diff
@@ -8,6 +8,58 @@ import logic.function.basic
 
 universe u
 
+section associative
+variables {α : Type u} (f : α → α → α) [is_associative α f] (x y : α)
+
+/--
+Composing two associative operations of `f : α → α → α` on the left
+is equal to an associative operation on the left.
+-/
+@[simp] lemma comp_assoc_left : (f x) ∘ (f y) = (f (f x y)) :=
+by { ext z, rw [function.comp_apply, @is_associative.assoc _ f] }
+
+/--
+Composing two associative operations of `f : α → α → α` on the right
+is equal to an associative operation on the right.
+-/
+@[simp] lemma comp_assoc_right : (λ z, f z x) ∘ (λ z, f z y) = (λ z, f z (f y x)) :=
+by { ext z, rw [function.comp_apply, @is_associative.assoc _ f] }
+
+end associative
+
+section semigroup
+variables {α : Type*}
+
+/--
+Composing two multiplications on the left by `y` then `x`
+is equal to a multiplication on the left by `x * y`.
+-/
+@[simp, to_additive] lemma comp_mul_left [semigroup α] (x y : α) :
+  ((*) x) ∘ ((*) y) = ((*) (x * y)) :=
+comp_assoc_left _ _ _
+
+/--
+Composing two additions on the left by `y` then `x`
+is equal to a addition on the left by `x + y`.
+-/
+add_decl_doc comp_add_left
```

Comment:
[You can specify the docstring as part of the call to `to_additive`](https://leanprover-community.github.io/mathlib_docs/attributes.html#to_additive). Though this works just as well, of course.

Reply:
Would you prefer the docstring to be set via the tag, or via the separate tactic?

Reply:
I lean very slightly towards using the tag since it saves a few lines. You won't have to give the name of the additive variation either, see e.g.: 

https://github.com/leanprover-community/mathlib/blob/aff7727008e59b8ea7ea30ccfae49282787ba0f4/src/data/equiv/transfer_instance.lean#L63-L71